lib/puppet/type/file/content.rb

```diff
@@ -185,6 +185,8 @@ def each_chunk_from(source_or_content)
         yield source_or_content
       elsif content_is_really_a_checksum? && source_or_content.nil?
         yield read_file_from_filebucket
+      elsif source_or_content.respond_to?(:metadata) && source_or_content.metadata.respond_to?(:source) && source_or_content.metadata.source =~ /^s3?:/
```

Comment:
Is there a better way to run this check?

My original check on the object was causing an exception due to not all source_or_content objects passed to the function having the property I needed to check.